Wine rework experiments

TheComputerGuy96 opened the pull request.

TheComputerGuy96 committed:
Remove 32-bit support

This will be unnecessary with my new Wine build (I have doubts about
--allow=multiarch though)

TheComputerGuy96 committed:
Remove the old Wine build

It won't probably work in this 32-bit-free environment (and I'll
make a better one anyway)

TheComputerGuy96 committed:
Add some Grapejuice patches

These are important for Grapejuice to function correctly now
(except for the winedebug one which is more for making debugging
easier)

guihkx commented:
I understand this is just a draft yet, but wouldn't it be preferable to upstream these patches instead?

TheComputerGuy96 commented:
> I understand this is just a draft yet, but wouldn't it be preferable to upstream these patches instead?

@guihkx One of them will definitely be PR'd (the Roblox launch fix) but the other two are controversial and will likely get rejected (so I don't really want to submit those)

The WebView2 one only really works because we control the Wine environment (maybe it can be PR'd with a Flatpak check though?)

As for the WINEDEBUG one I think it should do no harm in most cases (there are some edge cases but I couldn't properly investigate those)

TheComputerGuy96 committed:
Re-add --allow=multiarch permission

Wine in WoW64 mode still does one 32-bit syscall:
https://gitlab.winehq.org/wine/wine/-/blob/master/dlls/ntdll/unix/signal_x86_64.c#L106

Also fix a mistake in the Roblox launch fix patch (Roblox app should
launch now; Studio is still untested by me though)

TheComputerGuy96 committed:
Fix Roblox Studio executable path

Roblox Studio launch process is a bit weird
